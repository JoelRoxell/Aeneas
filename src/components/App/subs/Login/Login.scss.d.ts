// This file is generated automatically
export const login: string;
export const view: string;
export const title: string;
export const error: string;
export const registerButton: string;
