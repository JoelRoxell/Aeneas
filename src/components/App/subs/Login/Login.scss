.login {
  display: flex;
  flex-direction: column;
  align-items: center;
  flex: 1;
  background: url('/src/assets/img/leaf.jpg');

  .view {
    display: flex;
    flex-direction: column;
    margin-top: rem(85);
    margin-bottom: rem(45);
    background: white;
    width: rem(617);
    height: rem(710);
    box-shadow: 0 rem(2) rem(21) 0 rgba(0, 0, 0, 0.08);
    padding-left: rem(145);
    padding-right: rem(145);
    justify-content: center;

    .input {
      margin-bottom: rem(33);
    }

    .logo {
      margin-bottom: rem(41);
    }

    .form {
      width: rem(336);
      display: flex;
      flex-direction: column;
      align-items: center;
    }

    .title {
      color: black;
      text-align: center;
    }

    .error {
      color: $error;
      margin-top: rem(24);
    }
  }

  .split {
    display: flex;
    align-items: center;
    margin-top: rem(32);
    margin-bottom: rem(32);

    .line {
      width: rem(154);
      height: 1px;
      background-color: $light_gray;
    }

    .text {
      margin-right: rem(16);
      margin-left: rem(16);
    }
  }

  .copy {
    opacity: 0.8;
    color: #fff;
  }
}
