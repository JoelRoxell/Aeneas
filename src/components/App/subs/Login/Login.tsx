import * as React from 'react'

import Input from 'components/common/Input'
import Button from 'components/common/Button'

import * as style from './Login.scss'
import { UserContext } from 'contexts/user-context'
import Loader from 'components/common/Loader'

function Login() {
  const user = React.useContext(UserContext)
  console.log(user)

  return (
    <div className={style.login}>
      <form
        className={style.view}
        onSubmit={e => {
          e.preventDefault()

          user.signIn()
        }}
      >
        <div className={style.title}>{`Login title`}</div>

        <Input
          title="Email"
          name="email"
          placeholder="email"
          value={user.email}
          autoComplete="email"
          onChange={(name, value) => user.setUser({ ...user, [name]: value })}
        />

        <Input
          title="Password"
          name="password"
          placeholder="password"
          value={user.password}
          autoComplete="password"
          type="password"
          onChange={(name, value) => user.setUser({ ...user, [name]: value })}
        />

        <Button type="primary" title="Sign in" />

        <Button to="register" type="default" title="Register" className={style.registerButton} />

        {user.loading && <Loader />}

        {/* {error && <div className={style.error}>{error}</div>} */}
      </form>
    </div>
  )
}

export default Login
