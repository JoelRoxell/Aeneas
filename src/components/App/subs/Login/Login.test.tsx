import * as React from 'react'
import { mount } from 'enzyme'
import { BrowserRouter as Router } from 'react-router-dom'

import Login from 'components/App/subs/Login'
import UserProvider from 'contexts/user-context'

describe('Login', () => {
  test('render', () => {
    const wrapper = mount(
      <UserProvider>
        <Router>
          <Login />
        </Router>
      </UserProvider>
    )

    expect(wrapper).toMatchSnapshot()
  })
})
