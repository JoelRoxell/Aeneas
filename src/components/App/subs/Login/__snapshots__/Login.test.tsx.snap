// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login render 1`] = `
<UserProvider>
  <BrowserRouter>
    <Router
      history={
        Object {
          "action": "POP",
          "block": [Function],
          "createHref": [Function],
          "go": [Function],
          "goBack": [Function],
          "goForward": [Function],
          "length": 1,
          "listen": [Function],
          "location": Object {
            "hash": "",
            "pathname": "/",
            "search": "",
            "state": undefined,
          },
          "push": [Function],
          "replace": [Function],
        }
      }
    >
      <Login>
        <div
          className="login"
        >
          <ForwardRef
            pose="visible"
          >
            <PoseElement
              elementType="div"
              innerRef={null}
              pose="visible"
              poseConfig={
                Object {
                  "hidden": Object {
                    "opacity": 0,
                    "y": "3%",
                  },
                  "visible": Object {
                    "delayChildren": 300,
                    "opacity": 1,
                    "staggerChildren": 50,
                    "transition": Object {
                      "y": Object {
                        "damping": 50,
                        "stiffness": 800,
                        "type": "spring",
                      },
                    },
                    "y": "0%",
                  },
                }
              }
            >
              <div>
                <Card
                  className="view"
                >
                  <div
                    className="card view"
                  >
                    <form
                      className="form"
                      onSubmit={[Function]}
                    >
                      <Logo
                        className="logo"
                      >
                        <ForwardRef
                          pose="visible"
                        >
                          <PoseElement
                            elementType="div"
                            getInitialPoseFromParent={[Function]}
                            getParentPoseConfig={[Function]}
                            innerRef={null}
                            onUnmount={[Function]}
                            pose="visible"
                            poseConfig={
                              Object {
                                "hidden": Object {
                                  "opacity": 0,
                                  "y": "0%",
                                },
                                "visible": Object {
                                  "opacity": 1,
                                  "transition": Object {
                                    "default": Object {
                                      "duration": 500,
                                      "ease": "linear",
                                    },
                                    "opacity": Object {
                                      "duration": 600,
                                      "ease": "easeOut",
                                    },
                                    "stiffness": 100,
                                    "type": "spring",
                                    "y": Object {
                                      "duration": 600,
                                      "ease": "easeOut",
                                    },
                                  },
                                  "y": "-3%",
                                },
                              }
                            }
                            registerChild={[Function]}
                          >
                            <div>
                              <div
                                className="logo logo"
                              >
                                <Icon
                                  name="logo"
                                  svgStyle={
                                    Object {
                                      "fill": "#e2c085",
                                    }
                                  }
                                >
                                  <i
                                    className="icon"
                                  >
                                    <default
                                      fill="#e2c085"
                                    />
                                  </i>
                                </Icon>
                              </div>
                            </div>
                          </PoseElement>
                        </ForwardRef>
                      </Logo>
                      <Input
                        autoComplete="email"
                        className="input"
                        name="email"
                        onChange={[Function]}
                        placeholder="email"
                        rightIcon="user"
                        title="User"
                        validate={[Function]}
                        value=""
                      >
                        <div
                          className="input input"
                        >
                          <div
                            className="label"
                          >
                            User
                          </div>
                          <div
                            className="inputWrapper"
                          >
                            <input
                              autoComplete="email"
                              className="textField"
                              name="email"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              placeholder="email"
                              type="text"
                              value=""
                            />
                            <Icon
                              name="user"
                              svgStyle={
                                Object {
                                  "className": "rightIcon",
                                  "height": 24,
                                  "width": 24,
                                }
                              }
                            >
                              <i
                                className="icon"
                              >
                                <default
                                  className="rightIcon"
                                  height={24}
                                  width={24}
                                />
                              </i>
                            </Icon>
                          </div>
                        </div>
                      </Input>
                      <Input
                        autoComplete="password"
                        className="input"
                        name="password"
                        onChange={[Function]}
                        placeholder="password"
                        rightIcon="eye"
                        title="Password"
                        type="password"
                        validate={[Function]}
                        value=""
                      >
                        <div
                          className="input input"
                        >
                          <div
                            className="label"
                          >
                            Password
                          </div>
                          <div
                            className="inputWrapper"
                          >
                            <input
                              autoComplete="password"
                              className="textField"
                              name="password"
                              onBlur={[Function]}
                              onChange={[Function]}
                              onFocus={[Function]}
                              placeholder="password"
                              type="password"
                              value=""
                            />
                            <Icon
                              name="eye"
                              svgStyle={
                                Object {
                                  "className": "rightIcon",
                                  "height": 24,
                                  "width": 24,
                                }
                              }
                            >
                              <i
                                className="icon"
                              >
                                <default
                                  className="rightIcon"
                                  height={24}
                                  width={24}
                                />
                              </i>
                            </Icon>
                          </div>
                        </div>
                      </Input>
                      <Button
                        disabled={true}
                        title="Sign in"
                      >
                        <button
                          className="button primary disabled"
                          disabled={true}
                        >
                          <p>
                            Sign in
                          </p>
                        </button>
                      </Button>
                      <div
                        className="split"
                      >
                        <div
                          className="line"
                        />
                        <div
                          className="text"
                        >
                          or
                        </div>
                        <div
                          className="line"
                        />
                      </div>
                      <Button
                        title="Register as a new User"
                        to="register"
                        type="secondary"
                      >
                        <NavLink
                          activeClassName="activeLink"
                          className="button secondary"
                          to="register"
                        >
                          <Link
                            aria-current={null}
                            className="button secondary"
                            to="register"
                          >
                            <a
                              aria-current={null}
                              className="button secondary"
                              href="/register"
                              onClick={[Function]}
                            >
                              <div
                                className="inner"
                              >
                                <p
                                  className="text"
                                >
                                  Register as a new User
                                </p>
                              </div>
                            </a>
                          </Link>
                        </NavLink>
                      </Button>
                      <div
                        className="error"
                      />
                      <Loader
                        active={false}
                      >
                        <div
                          className="loader"
                        >
                          <PoseGroup
                            flipMove={true}
                          >
                            <Transition
                              enterAfterExit={false}
                              enterPose="enter"
                              exitPose="exit"
                              flipMove={true}
                              preEnterPose="exit"
                            />
                          </PoseGroup>
                        </div>
                      </Loader>
                    </form>
                  </div>
                </Card>
                <div
                  className="copy"
                >
                  © 2019 X AB. All rights reserved.
                </div>
              </div>
            </PoseElement>
          </ForwardRef>
        </div>
      </Login>
    </Router>
  </BrowserRouter>
</UserProvider>
`;
